Analysis/Tasks/PWGHF/HFCandidateCreator2Prong.cxx

```diff
@@ -179,6 +181,228 @@ struct HFCandidateCreator2ProngMC {
+  Produces<o2::aod::HfCandProng2FullParticles> rowCandidateFullParticles;
+  void init(InitContext const&)
+  {
+  }
+  void process(aod::Collisions const& collisions,
+               aod::McCollisions const& mccollisions,
+               soa::Join<aod::HfCandProng2, aod::HfCandProng2MCRec> const& candidates,
+               soa::Join<aod::McParticles, aod::HfCandProng2MCGen> const& particles,
+               aod::BigTracksPID const& tracks)
+  {
+    int ncol = 0;
+    rowCandidateFullEvents.reserve(collisions.size());
+    Printf("Collisions");
+    for (auto& collision : collisions) {
+      Printf("Collision %i", ncol++);
+      rowCandidateFullEvents(
+        collision.bcId(),
+        collision.numContrib(),
+        collision.posX(),
+        collision.posY(),
+        collision.posZ(),
+        0,
+        1);
+    }
+    int ncand = 0;
+    rowCandidateFull.reserve(candidates.size());
+    Printf("Candidates");
+    for (auto& candidate : candidates) {
+      // Printf("Candidate %i", ncand++);
+      rowCandidateFull(
+        candidate.index0_as<aod::BigTracksPID>().collision().bcId(),
+        candidate.index0_as<aod::BigTracksPID>().collision().numContrib(),
+        candidate.posX(),
+        candidate.posY(),
+        candidate.posZ(),
+        candidate.xSecondaryVertex(),
+        candidate.ySecondaryVertex(),
+        candidate.zSecondaryVertex(),
+        candidate.errorDecayLength(),
+        candidate.errorDecayLengthXY(),
+        candidate.chi2PCA(),
+        candidate.rSecondaryVertex(),
+        candidate.decayLength(),
+        candidate.decayLengthXY(),
+        candidate.decayLengthNormalised(),
+        candidate.decayLengthXYNormalised(),
+        candidate.impactParameterNormalised0(),
+        TMath::Abs(candidate.ptProng0()),
+        TMath::Sqrt(RecoDecay::P(candidate.pxProng0(), candidate.pyProng0(), candidate.pzProng0())),
+        candidate.impactParameterNormalised1(),
+        TMath::Abs(candidate.ptProng1()),
+        TMath::Sqrt(RecoDecay::P(candidate.pxProng1(), candidate.pyProng1(), candidate.pzProng1())),
+        candidate.pxProng0(),
+        candidate.pyProng0(),
+        candidate.pzProng0(),
+        candidate.pxProng1(),
+        candidate.pyProng1(),
+        candidate.pzProng1(),
+        candidate.impactParameter0(),
+        candidate.impactParameter1(),
+        candidate.errorImpactParameter0(),
+        candidate.errorImpactParameter1(),
+        candidate.index0_as<aod::BigTracksPID>().tpcNSigmaPi(),
+        candidate.index0_as<aod::BigTracksPID>().tpcNSigmaKa(),
+        candidate.index0_as<aod::BigTracksPID>().tofNSigmaPi(),
+        candidate.index0_as<aod::BigTracksPID>().tofNSigmaKa(),
+        candidate.index1_as<aod::BigTracksPID>().tpcNSigmaPi(),
+        candidate.index1_as<aod::BigTracksPID>().tpcNSigmaKa(),
+        candidate.index1_as<aod::BigTracksPID>().tofNSigmaPi(),
+        candidate.index1_as<aod::BigTracksPID>().tofNSigmaKa(),
+        InvMassD0(candidate),
```

Comment:
hi @njacazio @nzardosh @vkucera are we here only filling the D0mass?

Reply:
Yes, I can add the others if needed